### ISSUE TITLE
Setting host to "localhost" does NOT bind keepasshttp to localhost

### ISSUE THREAD

**Issue by schober** [pfn/keepasshttp#237]
So let's examine what keepasshttp bins to when we let the 'host' field be the default value, "localhost":

› sudo netstat -lnp | grep 19455                                                                  
tcp        0      0 0.0.0.0:19455           0.0.0.0:*               LISTEN      28660/cli    

Wait, it's bound to 0.0.0.0? So, what's the value if I set host to 127.0.0.1?

› sudo netstat -lnp | grep 19455
tcp        0      0 127.0.0.1:19455         0.0.0.0:*               LISTEN      29500/cli   

Now that's what I'm looking for. Okay, just to verify that the first result is undesirable, let's also test explicitly setting it to 0.0.0.0.

› sudo netstat -lnp | grep 19455
tcp        0      0 0.0.0.0:19455           0.0.0.0:*               LISTEN      24099/cli    

Yep, so the default "localhost" apparently binds to "world visible" 0.0.0.0, in spite of its intention. What if I set it to empty string? Okay, now I get some exception on startup claiming incompatibility between KP and KPH. Well, it actually progresses past here just fine, but... whatever.

**Comment by pfn** [comment]
It's a limitation of the .net http listener

--

**Comment by nifr** [comment]
So this sounds like a serious security issue that could be fixed by setting the default host 127.0.0.1, correct?

**Comment by pfn** [comment]
No, it is not a security issue. the .NET http listener will only send a
response for localhost requests. other requests get error 400.
